### PR TITLE
[@types/node] Add worker.SHARE_ENV and WorkerOptions.env

### PR DESCRIPTION
**Pull request by dnalborczyk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
- https://nodejs.org/dist/latest-v13.x/docs/api/worker_threads.html#worker_threads_worker_share_env
- https://nodejs.org/dist/latest-v13.x/docs/api/worker_threads.html#worker_threads_new_worker_filename_options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/node/worker_threads.d.ts [hunk 1]
```diff
@@ -54,6 +55,7 @@ declare module "worker_threads" {
     }
 
     interface WorkerOptions {
+        env?: NodeJS.ProcessEnv | typeof SHARE_ENV;
```
Comment (SimonSchick) [inline types/node/worker_threads.d.ts]: `typeof` shouldn't be needed here.

Comment (dnalborczyk) [reply types/node/worker_threads.d.ts]: I believe this is correct as is, if I remove `typeof` I get an exception: `'SHARE_ENV' refers to a value, but is being used as a type here.`

Comment (dnalborczyk) [reply types/node/worker_threads.d.ts]: doc reference: http://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#unique-symbol

Comment (SimonSchick) [reply types/node/worker_threads.d.ts]: Ah yea I misread, can you add a test for your additions tho? ❤️

**Review by SimonSchick — APPROVED** [review]

File: types/node/worker_threads.d.ts (hunk 1 above)
Comment (dnalborczyk) [reply types/node/worker_threads.d.ts]: No problem! I'll have a look.

**Comment by armanio123** [comment]
@SimonSchick do you want me to wait for the tests or should I merge this?

**Comment by dnalborczyk** [comment]
@SimonSchick @armanio123  I haven't found time yet to look at the tests and how they are structured. might need a couple more days.

**Comment by elibarzilay** [comment]
ping?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
Marked as changes requested until tests are added.

**Pull request closed without merging**